runtime/src/rpc.rs

```diff
@@ -243,39 +249,38 @@ impl InterBtcParachain {
 
         *next_nonce = ret.saturating_add(1);
 
-        ret
+        Ok(ret)
     }
 
-    async fn query_finalized<Address>(
-        &self,
-        address: Address,
-    ) -> Result<Option<<Address::Target as DecodeWithMetadata>::Target>, Error>
+    async fn query_finalized<Address>(&self, address: Address) -> Result<Option<Address::Target>, Error>
     where
         Address: StorageAddress<IsFetchable = Yes>,
     {
         let hash = self.get_finalized_block_hash().await?;
-        Ok(self.api.storage().fetch(&address, hash).await?)
+        match hash {
+            Some(hash) => Ok(self.api.storage().at(hash).fetch(&address).await?),
+            None => Ok(self.api.storage().at_latest().await?.fetch(&address).await?),
+        }
```

Comment:
Maybe we should change `get_finalized_block_hash` (to not return an `Option`) so this isn't necessary

Reply:
The `get_finalized_block_hash` returns the latest block for the test feature.